src/main/java/core/basesyntax/service/strategy/strategyimpl/PurchaseOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.strategy.strategyimpl;
+
+import core.basesyntax.dao.Storage;
+import core.basesyntax.model.FruitRecord;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(FruitRecord transaction) {
+        Integer currentQuantity = Storage.storage.get(transaction.getFruit());
+        int purchaseQuantity = transaction.getQuantity();
+        if (currentQuantity == null && currentQuantity < 0) {
```

Comment:
The condition `currentQuantity == null && currentQuantity < 0` will always be false because if `currentQuantity` is null, it cannot be compared to an int. It seems like you want to check if `currentQuantity` is null or less than 0, which should be done with two separate checks.